src/Model/Behavior/CryptBehavior.php

```diff
@@ -99,7 +99,7 @@ public function findDecrypted(Query $query, array $options)
             $driver = $this->_table->connection()->driver();
             foreach ($this->config('fields') as $field => $type) {
                 if (($options['fields'] && !in_array($field, (array)$options['fields']))
-                    || !$row->has($field)
+                    || is_string($row) || (is_array($row) ? !isset($row[$field]) : !$row->has($field))
```

Comment:
Shouldn't we check if $query->isHydrationEnabled() before any of this? Why add the formatter at all if it isn't enabled? ( I meant to add the comment here)